packages/suite/src/components/suite/modals/AdvancedCoinSettings/components/CustomBlockbookUrls.tsx

```diff
@@ -0,0 +1,149 @@
+    coin: Network['symbol'];
+    coinInfo: CoinInfo;
+    blockbookUrls: BlockbookUrl[];
+    addBlockbookUrl: (params: BlockbookUrl) => void;
+    removeBlockbookUrl: (params: BlockbookUrl) => void;
+}
+
+const CustomBlockbookUrls = ({
+    coin,
+    coinInfo,
+    blockbookUrls,
+    addBlockbookUrl,
+    removeBlockbookUrl,
+    ...props
+}: Props) => {
+    const [addErrorMessage, setAddErrorMessage] = useState<string | null>(null);
+    const addRef = useRef<HTMLInputElement>(null);
+
+    const addUrl = () => {
+        if (addRef.current !== null) {
+            const url = addRef.current.value;
+
+            // URL is not valid
+            if (!isUrl(url)) {
+                setAddErrorMessage(
+                    props.intl.formatMessage(messages.TR_CUSTOM_BACKEND_INVALID_URL),
+                );
+                return;
+            }
+
+            // URL already exists
+            if (blockbookUrls.find(b => b.coin === coin && b.url === url)) {
+                setAddErrorMessage(
+                    props.intl.formatMessage(messages.TR_CUSTOM_BACKEND_BACKEND_ALREADY_ADDED),
+                );
+                return;
+            }
+
+            // Add blockbook url
+            addBlockbookUrl({
+                coin,
+                url: addRef.current.value,
+            });
+
+            // Clear add form and errors
+            addRef.current.value = '';
+            setAddErrorMessage(null);
+        }
+    };
+
+    const urls = blockbookUrls.filter(b => b.coin === coin);
+
+    return (
+        <Wrapper>
+            <Heading>
+                <Translation id="SETTINGS_ADV_COIN_BLOCKBOOK_TITLE" />
+            </Heading>
+            <Description>
+                <Translation id="SETTINGS_ADV_COIN_BLOCKBOOK_DESCRIPTION" />
+            </Description>
+            <DefaultValues>
+                <Translation
+                    id="TR_DEFAULT_VALUE"
+                    values={{
+                        value: coinInfo.blockchainLink
+                            ? coinInfo.blockchainLink.url.join(', ')
+                            : '',
+                    }}
+                />
+            </DefaultValues>
+
+            {urls.map(b => (
+                <Input
+                    key={b.url}
+                    value={b.url}
+                    noTopLabel
+                    isDisabled
+                    innerAddon={
+                        <Button
+                            variant="tertiary"
+                            icon="CROSS"
+                            onClick={() => removeBlockbookUrl(b)}
+                        />
+                    }
+                />
+            ))}
+
+            <Input
+                placeholder={`e.g. https://${coin}1.trezor.io/`}
+                innerRef={addRef}
+                noTopLabel
+                type="text"
+                state={addErrorMessage ? 'error' : undefined}
+                bottomText={addErrorMessage}
```

Comment:
validation in `onChange` event,
you can get an inspiration from (simple react-hook-form implementation):
https://github.com/trezor/trezor-suite/blob/ffb9738f29e8686cd28e40ecb8c606d747ca1658/packages/suite/src/views/wallet/send/components/Raw/index.tsx